### PR TITLE
Ensure request body Content-Length is set correctly

### LINKED ISSUE #217: When body is not string next request on same host sends wrong method
```
local http = require "resty.http"
local client = http:new()
local res, err = client:request_uri("http://127.0.0.1", {
        method = "POST",
        body = 123213,
})
-- sends a POST request with empty body

local res, err = client:request_uri("http://127.0.0.1", {
        method = "DELETE",
})
-- sends a request with method line of "123213DELETE"
```

### PR DESCRIPTION
**Pull request by pintsized**
Fixes #217

### CODE REVIEW

**Review by hamishforbes — COMMENTED** [review]
I think just silently ignoring non-strings isn't ideal.
We should at least log a message that you're trying to send something that's being ignored.

Would it be better to try and convert to a string? Just with `tostring`, no serialising or anything.
If you set body to a number or a bool its reasonable to expect that to work and just send a stringified version? Certainly for numbers anyway.

**Comment by pintsized** [comment]
I guess I was of the opinion that the API is perfectly clear (as is `tcpsock:send` incidentally so I'm surprised it allows Lua numbers) but we should do something because it's possible to get the client into a weird state. I just wanted to make the minimal change possible really.

But yeah, it's perhaps more idiomatic to assume we'll send string representations magically, and I guess that's harmless enough. I wouldn't be surprised if LuaJIT elides `tostring(string)` where possible anyway.

**Comment by pintsized** [comment]
On closer inspection this is actually about not having a `Content-Length` sent in the request, and so the body is left on the wire. Also `tcpsock:send` supports a table of strings so I added explicit support for that as well.

I also thought I'd be good and return an error if your body is an iterator and you haven't set a length. But I'd forgotten about the `get_client_body_reader` in chunked transfer mode. Test coverage appears to be non-existent here. @hamishforbes Do you remember anything about this? Is it a supported thing - the comment suggests maybe not? Perhaps we need a flag the user must set, to treat the request body iterator as chunked or not? Or we drop my error handling and just assume chunked if there's no length?

This all feels like some foggy distant memory so any thoughts appreciated!

File: lib/resty/http.lua [hunk 1]
```diff
@@ -398,6 +398,25 @@ local function _receive_headers(sock)
...
+        te = tbl_concat(te, "")
+    end
+
+    local ok, encoding = pcall(str_lower, te)
+    if not ok then
+        encoding = ""
+    end
+
+    return str_find(encoding, "chunked", 1, true) ~= nil
+end
```
Comment (Tieske) [inline lib/resty/http.lua:416]:
```suggestion
local function transfer_encoding_is_chunked(headers)
    local te = headers["Transfer-Encoding"]
    if not te then
        return false
    end

    -- Handle duplicate headers
    -- This shouldn't happen but can in the real world
    if type(te) ~= "string" then
        te = tbl_concat(te, ",")
    end

    return str_find(str_lower(te), "chunked", 1, true) ~= nil
end
```

changes:
- exit early if not found
- check against "string" type, the table concat will fail hard if it isn't a table but that is ok
- table concat, using a comma `,` to ensure proper header semantics
- skip the pcall, no longer necessary

**Review by hamishforbes — APPROVED** [review]

**Pull request merged**